File: Solution with Js/189.js
const revNums = (nums, start, end) => {
    while(start<end){
        [nums[start], nums[end]] = [nums[end], nums[start]];
        start++;
        end--;
    }
}

var rotate = function(nums, k){
    k = k%nums.length;
    nums.reverse();
    revNums(nums,0,k-1);
    revNums(nums,k,nums.length-1);
    return nums;
};
console.log(rotate([1,2,3,4,5,6,7],3))

File: Solution with python/189.py
class Solution:
    def rotate(self,nums,k):
        n=len(nums)
        k = k%n 
        nums[:n-k]= nums[:n-k][::-1]
        nums[n-k:]=nums[n-k:][::-1]
        nums[:]=nums[::-1]
        return nums
    
    
a = Solution()
print(a.rotate([1,2,3,4,5,6,7],3))